src/devices/ant_antplus.c

```diff
@@ -0,0 +1,164 @@
+network key and defined device types, each sending "data pages" of 8 bytes with
+specific data for the device type. Most ANT+ devices are sports focused like
+heart rate monitors, bicycle sensors, or environmental sensors.
+
+Please note that unlike most devices in the rtl_433 repository, ANT+ devices
+operate in the ISM band between 2.4GHz and 2.5GHz. Decoding these signals
+requires an SDR capable of operating above 2.4GHz (e.g. PlutoSDR) or the
+use of a downconverter for rtl_sdr. The ISM band is very noisy, so it's
+recommended to get the device very close, use only a ~30mm wire antenna and
+use mid-low gains. Finally, since the protocol encodes at 1Mbps, sampling
+rate should be -s 4M or higher. It can work with a sampling rate as low as
+2Msps, but unreliably. The following works well with PlutoSDR:
+
+rtl_433 -d driver=plutosdr,uri=ip:192.168.2.1 -g 20 -f 2457.025M -s 4M
+
+Without knowing the 8 byte key, existing ANT chips do not allow sniffing
+of the data packets. Hence the need for this decoder. At the moment it's
+not possible to recover the ANT key from the 16 bit on-air network key.
+
+This decoder only captures and displays the low level packets, identifying the
+2 byte network key plus all other device characteristics and 8 byte ANT payload.
+It identifies ANT+ packets using the unique network key used (0xa6c5)
+Refer to ANT+ documentation for ech specific device to parse the content of
+the pages sent as 8 byte ANT+ payload.
+
+The ANT protocol is using an uncommon strategy for the premable: either 0x55 or
+0xaa, depending on the value of the first bit of the following byte (0 and 1
+respectively). The nRF24L01, on which the ANT protocol is based, uses the same
+premable strategy. In order to determine if the packet is using 0x55 or 0xaa,
+both packets are extracted and the only valid one is determined by matching
+the packet CRC with a calculated CRC value for both alternative packets.
+
+The payload is 18 bytes long structured as follows:
+
+    PNNDDTXLPPPPPPPPCC
+
+- P: Preamble: either 0x55 or 0xAA, depending on the value of first bit of the next byte
+- N: Network key, assume LSB first (ANT+ uses 0xc5a6, most invalid keys 0x255b)
+- D: Device number, 16 bit. LSB first
+- X: Transmission type
+- L: ANT payload length including CRC
+- P: 8 byte ANT or ANT+ payload
+- C: 16 bit CRC (CRC-16/CCITT-FALSE)
+
+Weirdly, L is always 10, because at the moment the ANT payload is always 8 bytes
+
+CREDITS:
+https://github.com/sghctoma/antfs-poc-defcon24
+https://reveng.sourceforge.io/ to reverse engineer the CRC algorithm used
+*/
+
+#include "decoder.h"
+#include "util.h"
+#include "stdbool.h"
+
+static int ant_antplus_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble[] = {0xAA};
+    uint8_t b[17]; // aligned packet data for both preambles/offsets
+    unsigned bit_offset;
+    char payload[8 * 3 + 1]; // payload is 8 hex pairs for ANT and ANT+
+    bool antplus_flag = false;
+
+    data_t *data;
```

Comment:
put this on the data_make() line: `data_t *data = data_make(`

Reply:
will do